File: samples/sandbox/app/src/main/java/com/scichart/docsandbox/examples/javaBuilder/stylingAndTheming/PenStyle_BrushStyle_FontStyle.java
package com.scichart.docsandbox.examples.javaBuilder.stylingAndTheming;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Color;
import android.graphics.Typeface;

import androidx.annotation.NonNull;

import com.scichart.charting.visuals.SciChartSurface;
import com.scichart.docsandbox.R;
import com.scichart.docsandbox.core.ExampleDefinition;
import com.scichart.docsandbox.examples.base.SingleChart2DFragment;
import com.scichart.drawing.common.FontStyle;
import com.scichart.drawing.common.LinearGradientBrushStyle;
import com.scichart.drawing.common.LinearGradientPenStyle;
import com.scichart.drawing.common.PenStyle;
import com.scichart.drawing.common.RadialGradientBrushStyle;
import com.scichart.drawing.common.RadialGradientPenStyle;
import com.scichart.drawing.common.SolidBrushStyle;
import com.scichart.drawing.common.TextureBrushStyle;
import com.scichart.drawing.common.TexturePenStyle;

@ExampleDefinition()
public class PenStyle_BrushStyle_FontStyle extends SingleChart2DFragment {
    @Override
    protected void initExample(@NonNull SciChartSurface surface) {
        createPen();
        createBrush();
        createFontStyle();
    }

    void createPen() {
        final LinearGradientBrushStyle linearGradientBrush = new LinearGradientBrushStyle(0, 0, 0, 1, Color.RED, Color.BLUE);
        final RadialGradientBrushStyle radialGradientBrush = new RadialGradientBrushStyle(0.5f, 0.5f, 0.25f, 0.5f, Color.RED, Color.BLUE);

        final Bitmap texture = BitmapFactory.decodeResource(getResources(), R.drawable.example_scichartlogo);
        final TextureBrushStyle textureBrush = new TextureBrushStyle(texture);

        // <CreatePen>
        // Solid Pen
        PenStyle solidPenStyle = sciChartBuilder.newPen()
                .withColor(0xFF99EE99)
                .withAntiAliasing(true)
                .withThickness(1f)
                .withStrokeDashArray(new float[]{10, 3, 10, 3})
                .build();

        // Linear Gradient Pen requires LinearGradientBrushStyle
        final LinearGradientPenStyle linearGradientPenStyle = new LinearGradientPenStyle(linearGradientBrush, true, 1f, null);

        // Radial Gradient Pen requires RadialGradientBrushStyle
        final RadialGradientPenStyle radialGradientPenStyle = new RadialGradientPenStyle(radialGradientBrush, true, 1f, null);

        // Texture Pen requires TextureBrushStyle
        final TexturePenStyle texturedPenStyle = new TexturePenStyle(textureBrush, true, 1f, null);
        // </CreatePen>
    }

    void createBrush() {
        final Bitmap texture = BitmapFactory.decodeResource(getResources(), R.drawable.example_scichartlogo);

        // <CreateBrush>
        // Solid Brush
        final SolidBrushStyle solidBrushStyle = new SolidBrushStyle(Color.RED);

        // Linear Gradient Brush
        final LinearGradientBrushStyle linearGradientBrush = new LinearGradientBrushStyle(0, 0, 0, 1, Color.RED, Color.BLUE);

        // Radial Gradient Brush
        final RadialGradientBrushStyle radialGradientBrush = new RadialGradientBrushStyle(0.5f, 0.5f, 0.25f, 0.5f, Color.RED, Color.BLUE);

        // Texture Brush - Bitmap with Texture should be provided
        final TextureBrushStyle textureBrush = new TextureBrushStyle(texture);
        // </CreateBrush>
    }

    void createFontStyle() {
        // <CreateFontStyle>
        Typeface typeface = Typeface.createFromAsset(getActivity().getAssets(), "font/font.ttf");
        FontStyle fontStyle = sciChartBuilder.newFont()
                .withTypeface(typeface)
                .withTextSize(14)
                .withTextColor(Color.RED)
                .withAntiAliasing(true)
                .build();
        // </CreateFontStyle>
    }
}

File: samples/sandbox/app/src/main/java/com/scichart/docsandbox/examples/kotlin/stylingAndTheming/PenStyle_BrushStyle_FontStyle.kt
package com.scichart.docsandbox.examples.kotlin.stylingAndTheming

import android.graphics.BitmapFactory
import android.graphics.Color
import android.graphics.Typeface
import com.scichart.charting.visuals.SciChartSurface
import com.scichart.docsandbox.R
import com.scichart.docsandbox.core.ExampleDefinition
import com.scichart.docsandbox.examples.base.SingleChart2DFragment
import com.scichart.drawing.common.*
import com.scichart.drawing.utility.ColorUtil

@ExampleDefinition()
class PenStyle_BrushStyle_FontStyle : SingleChart2DFragment() {
    override fun initExample(surface: SciChartSurface) {
        createPen()
        createBrush()
        createFontStyle()
    }

    fun createPen() {
        val linearGradientBrush = LinearGradientBrushStyle(0f, 0f, 0f, 1f, Color.RED, Color.BLUE)
        val radialGradientBrush =
            RadialGradientBrushStyle(0.5f, 0.5f, 0.25f, 0.5f, Color.RED, Color.BLUE)
        val texture = BitmapFactory.decodeResource(resources, R.drawable.example_scichartlogo)
        val textureBrush = TextureBrushStyle(texture)

        // <CreatePen>
        // Solid Pen
        val solidPenStyle = SolidPenStyle(-0x661167, true, 1f, floatArrayOf(10f, 3f, 10f, 3f))

        // Linear Gradient Pen requires LinearGradientBrushStyle
        val linearGradientPenStyle = LinearGradientPenStyle(linearGradientBrush, true, 1f, null)

        // Radial Gradient Pen requires RadialGradientBrushStyle
        val radialGradientPenStyle = RadialGradientPenStyle(radialGradientBrush, true, 1f, null)

        // Texture Pen requires TextureBrushStyle
        val texturedPenStyle = TexturePenStyle(textureBrush, true, 1f, null)
        // </CreatePen>
    }

    fun createBrush() {
        val texture = BitmapFactory.decodeResource(resources, R.drawable.example_scichartlogo)

        // <CreateBrush>
        // Solid Brush
        val solidBrushStyle = SolidBrushStyle(Color.RED)

        // Linear Gradient Brush
        val linearGradientBrush = LinearGradientBrushStyle(0f, 0f, 0f, 1f, Color.RED, Color.BLUE)

        // Radial Gradient Brush
        val radialGradientBrush = RadialGradientBrushStyle(0.5f, 0.5f, 0.25f, 0.5f, Color.RED, Color.BLUE)

        // Texture Brush - Bitmap with Texture should be provided
        val textureBrush = TextureBrushStyle(texture)
        // </CreateBrush>
    }

    fun createFontStyle() {
        // <CreateFontStyle>
        val typeface = Typeface.createFromAsset(requireActivity().assets, "font/font.ttf")
        val fontStyle = FontStyle(typeface, 14f, ColorUtil.Red, true)
        // </CreateFontStyle>
    }
}
